src/components/MTableBodyRow/index.js

```diff
@@ -49,17 +50,24 @@ function MTableBodyRow({ forwardedRef, ...props }) {
   const columns = propColumns.filter((columnDef) => !columnDef.hidden);
 
   const onClick = (event, callback) =>
-    callback(event, data, (panelIndex) => {
-      let panel = detailPanel;
-      if (Array.isArray(panel)) {
-        panel = panel[panelIndex || 0];
-        if (typeof panel === 'function') {
-          panel = panel(data);
+    callback(
+      event,
+      data,
+      (panelIndex) => {
+        let panel = detailPanel;
+        if (Array.isArray(panel)) {
+          panel = panel[panelIndex || 0];
+          if (typeof panel === 'function') {
+            panel = panel(data);
+          }
+          panel = panel.render;
         }
-        panel = panel.render;
+        onToggleDetailPanel(path, panel);
+      },
+      () => {
```

Comment:
Can we make this into an action obejct to make it more clearer? The two callbacks?